app/models/mixins/authentication_mixin.rb

```diff
@@ -158,14 +158,20 @@ def update_authentication(data, options = {})
     # Invoke before callback
     before_update_authentication if self.respond_to?(:before_update_authentication) && options[:save]
 
+    is_amazon = kind_of?(ManageIQ::Providers::Amazon::CloudManager)
```

Comment:
Hm really want to avoid this kind of thing in core

Reply:
Another option - just remove this condition, and pass `service_account` for everything. It might work, since we have specific kind of authentication and know that this field doesn't used in it (only in google's other type).

Reply:
:+1: yeah each provider should know what component they need to use

Reply:
Plus the UI should only be sending service_account for supported providers

Reply:
Yes, it is in UI:
* https://github.com/ManageIQ/manageiq-ui-classic/pull/5621/files#diff-002200c01a8a53ff2a4439390e6bc3c4R366
* https://github.com/ManageIQ/manageiq-ui-classic/pull/5621/files#diff-f402e411cd6dc158c724b14d72ead612R361